pkg/cloudclient/gcp/private.go

```diff
@@ -187,7 +205,26 @@ func (c *Client) ensureDNSForService(kclient client.Client, svc *corev1.Service,
 	return nil
 }
 
-func (c *Client) removeDNSForService(kclient client.Client, svc *corev1.Service, dnsName string) error {
+// Returns GCP forwarding rule for given IP
+func findGCPForwardingRuleForExtIP(c *GCPClient, rhapiLbIP string) (*compute.ForwardingRule, error) {
+	listCall := c.computeService.ForwardingRules.List(c.projectID, c.region)
+	response, err := listCall.Do()
+	if err != nil {
+		return nil, cioerrors.ForwardingRuleNotFound(err.Error())
+	}
+	var fr *compute.ForwardingRule
+	for _, lb := range response.Items {
+		if lb.IPAddress == rhapiLbIP {
+			fr = lb
+		}
+	}
+	if fr == nil {
+		return nil, fmt.Errorf("Forwarding rule not found in GCP for given service IP %s", rhapiLbIP)
+	}
+	return fr, nil
+}
```

Comment:
```suggestion
func (c *GCPClient) findGCPForwardingRuleForExtIP(rhapiLbIP string) (compute.ForwardingRule, error) {
	listCall := c.computeService.ForwardingRules.List(c.projectID, c.region)
	response, err := listCall.Do()
	if err != nil {
		return nil, err
	}

	for _, lb := range response.Items {
		if lb.IPAddress == rhapiLbIP {
			return lb, nil
		}
	}
	
	return compute.ForwardingRule{}, nil
}
```

Reply:
I'm not able to commit it as suggestion here, so I'll push the update.

Reply:
done